### PR TITLE
 Make index_hit_rate, index_caching and index_usage methods Citus compatible

### PR DESCRIPTION
**Pull request by naisila**
I have modified `index_hit_rate`, `index_caching` and `index_usage` with an `if else` statement depending on `citus_enabled` variable. I am mostly running command on placements for distributed tables. The method currently being used in the project is `index_hit_rate`.

### CODE REVIEW

File: lib/pghero/methods/indexes.rb [hunk 1]
```diff
@@ -1,49 +1,237 @@
...
+            WITH dist_indexes_stats AS(
+              WITH dist_indexes AS (
+                SELECT
+                  indexname,
+                  logicalrelid,
+                  schemaname,
+                  tablename
+                FROM
+                  pg_indexes
+                  JOIN pg_dist_partition ON tablename = logicalrelid::name OR schemaname::text || '.' || tablename::text = logicalrelid::text
```
Comment (byucesoy) [inline lib/pghero/methods/indexes.rb]: `ON (schemaname || '.' || tablename)::regclass = logicalrelid;` would be simpler.

File: lib/pghero/methods/indexes.rb [hunk 2]
```diff
@@ -1,49 +1,237 @@
...
+              relname AS table,
+              indexrelname AS index,
+              CASE WHEN idx_blks_hit + idx_blks_read = 0 THEN
+                0
+              ELSE
+                ROUND(1.0 * idx_blks_hit / (idx_blks_hit + idx_blks_read), 2)
+              END AS hit_rate
+            FROM (
+              SELECT
+                *
```
Comment (byucesoy) [inline lib/pghero/methods/indexes.rb]: Similar with the way I mentioned in the other PR, we can simplify moving above query to here.

File: lib/pghero/methods/indexes.rb [hunk 3]
```diff
@@ -1,49 +1,237 @@
...
+            FROM
+              pg_statio_user_indexes
+          SQL
+        else
+          select_one <<-SQL
+            WITH worker_idx_blks_stats_json AS (
+              SELECT
+                result::json
+              FROM
+                run_command_on_workers($cmd$
```
Comment (byucesoy) [inline lib/pghero/methods/indexes.rb]: Please check all instances of $cmd$ in the codebase and replace them with $$.

File: lib/pghero/methods/tables.rb [hunk 4]
```diff
@@ -1,67 +1,294 @@
...
+
       def table_hit_rate
-        select_one(<<-SQL
-          SELECT
-            sum(heap_blks_hit) / nullif(sum(heap_blks_hit) + sum(heap_blks_read), 0) AS rate
-          FROM
-            pg_statio_user_tables
-        SQL
-        )
+        if !citus_enabled?
```
Comment (byucesoy) [inline lib/pghero/methods/tables.rb]: Isn't these changes same with the changes in PR#5?

**Review by byucesoy — CHANGES_REQUESTED** [review]

**Comment by naisila** [comment]
@byucesoy I have implemented the changes.

File: lib/pghero/methods/indexes.rb [hunk 5]
```diff
@@ -1,49 +1,233 @@
...
+              relname ASC
+           SQL
+        else
+          select_all <<-SQL
+            WITH dist_tables_with_indexes (tablename) AS (
+              SELECT 
+                DISTINCT indrelid::regclass
+              FROM
+                pg_index
+                JOIN pg_dist_partition ON indrelid = logicalrelid
```
Comment (byucesoy) [inline lib/pghero/methods/indexes.rb:167]: JOINs should be aligned with FROM. (We should have fixed this in previous PRs too)

**Review by byucesoy — APPROVED** [review]
Fix the indentation and :shipit:

File: lib/pghero/methods/indexes.rb (hunk 5 above)
Comment (naisila) [reply lib/pghero/methods/indexes.rb:167]: 😢

Comment (byucesoy) [reply lib/pghero/methods/indexes.rb:167]: Just open another PR for all indentation issues, we can quickly merge it. No need for extensive reviews.

**Pull request merged**